File: frontend/src/lib/types/stock-validation.ts
export interface StockValidationRequest {
	items: {
		productId: number;
		quantity: number;
	}[];
}

export interface InsufficientIngredient {
	ingredientId: number;
	ingredientName: string;
	required: number;
	available: number;
	shortage: number;
	unit: string;
}

export interface StockValidationResponse {
	valid: boolean;
	insufficientStock: InsufficientIngredient[];
}

File: backend/internal/domain/stock_validation.go
package domain

// StockValidationRequest representa o payload para validação de estoque
type StockValidationRequest struct {
	Items []OrderItem `json:"items"`
}

// StockValidationResponse representa o resultado da validação
type StockValidationResponse struct {
	Valid             bool                     `json:"valid"`
	InsufficientStock []InsufficientIngredient `json:"insufficientStock,omitempty"`
}

// InsufficientIngredient representa um ingrediente com estoque insuficiente
type InsufficientIngredient struct {
	IngredientID   uint    `json:"ingredientId"`
	IngredientName string  `json:"ingredientName"`
	Required       float64 `json:"required"`
	Available      float64 `json:"available"`
	Shortage       float64 `json:"shortage"`
	Unit           string  `json:"unit"`
}
